onboarding-validation-keys-generator/main.go

```diff
@@ -81,7 +81,7 @@ func main() {
 		klog.Exitf("failed to delete public secret: %v", err)
 	}
 
-	err = controllerutil.SetOwnerReference(storageCluster, privateSecret, cl.Scheme())
+	err = controllerutil.SetControllerReference(storageCluster, privateSecret, cl.Scheme())
```

Comment:
Having the token generator add a controller ref for storage cluster is an antipattern. Why are we doing this? What prevents us having code in the storage cluster controller that take ownership on these secrets?